File: go/config/DialUp.go
package config

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// DialUp 对应 config/DialUp.py。
type DialUp struct {
	*session.ApiGroup
}

// NewDialUp 创建 DialUp API 分组。
func NewDialUp(s *session.Session) *DialUp {
	return &DialUp{ApiGroup: session.NewApiGroup(s)}
}

// Config 对应 dialup/config.xml。
func (d *DialUp) Config() (map[string]interface{}, error) {
	return d.S.Get("dialup/config.xml", nil, "config")
}

// Connectmode 对应 dialup/connectmode.xml。
func (d *DialUp) Connectmode() (map[string]interface{}, error) {
	return d.S.Get("dialup/connectmode.xml", nil, "config")
}

// Profileswitch 对应 dialup/profileswitch.xml。
func (d *DialUp) Profileswitch() (map[string]interface{}, error) {
	return d.S.Get("dialup/profileswitch.xml", nil, "config")
}

// LmtAutoModeDisconnect 对应 dialup/lmt_auto_mode_disconnect.xml。
func (d *DialUp) LmtAutoModeDisconnect() (map[string]interface{}, error) {
	return d.S.Get("dialup/lmt_auto_mode_disconnect.xml", nil, "config")
}

File: src/config/dial_up.rs
//! DialUp config group (`config/DialUp.py`).

use serde_json::Value;

use crate::api::get_value;
use crate::connection::Connection;
use crate::errors::Result;

/// DialUp config group.
pub struct DialUpConfig<'a> {
    conn: &'a Connection,
}

impl<'a> DialUpConfig<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        DialUpConfig { conn }
    }

    /// `dialup/config.xml`. Dial-up configuration.
    pub fn config(&self) -> Result<Value> {
        get_value(self.conn, "config/dialup/config.xml")
    }

    /// `dialup/connectmode.xml`. Dial-up connect mode.
    pub fn connectmode(&self) -> Result<Value> {
        get_value(self.conn, "config/dialup/connectmode.xml")
    }

    /// `dialup/profileswitch.xml`. Dial-up profile switch.
    pub fn profileswitch(&self) -> Result<Value> {
        get_value(self.conn, "config/dialup/profileswitch.xml")
    }

    /// `dialup/lmt_auto_mode_disconnect.xml`. Auto-mode disconnect status.
    pub fn lmt_auto_mode_disconnect(&self) -> Result<Value> {
        get_value(self.conn, "config/dialup/lmt_auto_mode_disconnect.xml")
    }
}
